tiledb/sm/storage_manager/context_resources.h

```diff
@@ -91,11 +105,21 @@ class ContextResources {
     return vfs_;
   }
 
+  [[nodiscard]] inline shared_ptr<RestClient> rest_client() const {
+    return rest_client_;
+  }
+
  private:
   /* ********************************* */
   /*         PRIVATE ATTRIBUTES        */
   /* ********************************* */
 
+  /** The configuration for this ContextResources */
+  mutable Config config_;
+
+  /** The class logger. */
+  shared_ptr<Logger> logger_;
```

Comment:
Just FYI the current `logger` design also follows a tree structure, somehow like `stats`, so we'll soon need to think about how we are going to handle that within ContextResources class.

Reply:
Ooof. Reading how that works its going to be more of an issue than the stats object even since its got dynamic tags being generated and propagated between instances.

Reply:
And thanks for the heads up! I don't think it changes anything for this PR, but its definitely something that'll need to be considered if/when ContextResources instances are passed further down.

Reply:
Also confirmed that we're using this in some places. Specifically, Query clones the logger off StorageManager, and then passes its cloned version to various Reader instances that clone the clone. Curiously we don't go through Array there as it seems like we'd want to know which queries came from which array.

Reply:
I think the only reason it's not in Array it's just that we didn't continue expanding the use of that hierarchical logger to all the classes of interest, logger work got de-prioritized at some point and then forgotten. I recall having the ambition to refactor all this to use something like a base class named "Observable", sort of an Interface, that classes interested in logger, stats, counter metrics etc would implement/inherit from and get for free the right logger, stats etc instance. I might have some rotten branch somewhere with that work, but I don't think it's relevant anymore.